src/Raven.Server/Documents/DatabasesLandlord.cs

```diff
@@ -874,41 +895,49 @@ public void UnloadDirectly(StringSegment databaseName, DateTime? wakeup = null,
 
             LastRecentlyUsed.TryRemove(databaseName, out _);
             DatabasesCache.RemoveLockAndReturn(databaseName.Value, CompleteDatabaseUnloading, out _, caller).Dispose();
-            ScheduleDatabaseWakeup(databaseName.Value, dueTime);
+
+            if (dueTime == 0)
+                return;
+
+            _wakeupTimers.TryAdd(databaseName.Value, new Timer(_ => StartDatabaseOnTimer(databaseName.Value, wakeup), null, dueTime, Timeout.Infinite));
```

Comment:
Why do not use `RescheduleDatabaseWakeup`? Is there a chance that we will have old timer that we need to dispose?

Reply:
UnloadDirectly() is called if and only if the DB become idle so _wakeupTimers cannot have the DB. RescheduleDatabaseWakeup called from putServerWideBackup only If the DB is idle, so it could have a timer already.